README.md

```diff
@@ -122,6 +122,46 @@ require('react-dom').render(reactElement, document.getElementById('root'));
 // <div><span style="font-size: 42px;">replaced!</span></div>
 ```
 
+Advanced usage with replace, keeping children (useful if child elements also needs to be replaced or just kept):
+
+```js
+var Parser = require('html-react-parser');
+var domToReact = require('html-react-parser/lib/dom-to-react'); // used for recursively parsing DOM created from the HTML
+var React = require('react');
+
+var html = '<div><p id="main"><span class="prettify">keep me and make me pretty!</span></p></div>';
+
+var parserConfig = {
+    replace: function(domNode, key) {
```

Comment:
I ended up removing `key` parameter in #18 as I realized there's a better and easier way of doing things.

If you would be so kind, could you remove `key` in your examples before performing the squash? Thanks so much for dealing with these changes and requests.

After your PR gets merged, I plan to do a release.